src/app/pages/questions/services/finish-task.service.ts

```diff
@@ -59,12 +59,14 @@ export class FinishTaskService {
 
   processQuestionnaireData(answers, timestamps, questions) {
     this.logger.log('Answers to process', answers)
-    const values = Object.entries(answers).map(([key, value]) => ({
-      questionId: { string: key.toString() },
-      value: { string: value.toString() },
-      startTime: timestamps[key].startTime,
-      endTime: timestamps[key].endTime
-    }))
+    const values = Object.entries(answers)
+      .filter(([k, v]) => timestamps[k])
```

Comment:
what is the case when a timestamp be null?

Reply:
I don't think there will be a case in production, but when I was testing earlier the matrix radio input were all submitting answers by default (so even those that are skipped with branching logic had answers) so there were answers even when there were no timestamps (because they were skipped). So I added this filter to make sure only those with non-null timestamps are included. Also I added a fix so that only those questions that are not skipped will be submitting default answers.

Reply:
ok cool. and a timestamp for default answers will be taken from the timestamp of the questionnaire?

Reply:
Yes when they start viewing the question.